sonic_platform_base/chassis_base.py

```diff
@@ -263,12 +272,57 @@ def get_module_index(self, module_name):
         Args:
             module_name: A string, prefixed by SUPERVISOR, LINE-CARD or FABRIC-CARD
             Ex. SUPERVISOR0, LINE-CARD1, FABRIC-CARD5
+            SmartSwitch Example: DPU0, DPU1, DPU2 ... DPUX
 
         Returns:
             An integer, the index of the ModuleBase object in the module_list
         """
         raise NotImplementedError
 
+    ##############################################
+    # SmartSwitch methods
+    ##############################################
+
+    def get_dpu_id(self, name):
+        """
+        Retrieves the DPU ID for the given dpu-module name.
+        Returns None for non-smartswitch chassis.
+
+        Returns:
+            An integer, indicating the DPU ID Ex: name:DPU0 return value 0,
+            name:DPU1 return value 1, name:DPUX return value X
+        """
+        raise NotImplementedError
+
+    def is_smartswitch(self):
+        """
+        Retrieves whether the sonic instance is part of smartswitch
+
+        Returns:
+            Returns:True for SmartSwitch and False for other platforms
+        """
+        return False 
+
+    def get_module_dpu_data_port(self, index):
```

Comment:
@rameshraghupathy We have a `role` type to indicate DPU data port. You can define this in port_confi.ini or hwsku.json 
Example https://github.com/sonic-net/sonic-buildimage/blob/master/device/mellanox/x86_64-nvidia_sn4280-r0/ACS-SN4280/hwsku.json#L125

Can we make use of hwsku.json or port_config.ini? Lets keep one approach. I am not sure if this hwsku.json/port_confi.ini is already mentioned in HLD

Reply:
I'm using hwsku.json

Reply:
@rameshraghupathy can we then remove this API as we follow suggested HLD approach?

Reply:
Removed